### PR TITLE
Media lib add allowed types to [CTB]

### PR DESCRIPTION
**Pull request by soupette**
#### Description of what you did:
This PR adds the option to specify the allowed file types for a field in the content type builder plugin.

### CODE REVIEW

File: packages/strapi-plugin-content-type-builder/admin/src/components/AllowedTypesSelect/MenuList.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+            {options.map(({ name, infos }) => {
+              const isChecked = value.value.includes(name);
+              const target = { name, value: !isChecked };
+
+              return (
+                <li key={name}>
+                  <CheckboxWrapper>
+                    <Label
+                      htmlFor={name}
+                      onClick={() => {
```
Comment (HichamELBSI) [inline packages/strapi-plugin-content-type-builder/admin/src/components/AllowedTypesSelect/MenuList.js]: Useless brackets

File: packages/strapi-plugin-content-type-builder/admin/src/components/AllowedTypesSelect/MenuList.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+};
+
+MenuList.propTypes = {
+  selectProps: PropTypes.shape({
+    changeMediaAllowedTypes: PropTypes.func.isRequired,
+    value: PropTypes.object,
+  }),
+};
+
+export default MenuList;
```
Comment (HichamELBSI) [inline packages/strapi-plugin-content-type-builder/admin/src/components/AllowedTypesSelect/MenuList.js]: There is a lot of inline style customization here. We should avoid this for the app performance and for the scalability.

File: packages/strapi-plugin-content-type-builder/admin/src/components/AllowedTypesSelect/MenuList.js (hunk 1 above)
Comment (soupette) [reply packages/strapi-plugin-content-type-builder/admin/src/components/AllowedTypesSelect/MenuList.js]: Let's keep them this is why I have removed this [rule](https://eslint.org/docs/rules/arrow-body-style)

File: packages/strapi-plugin-content-type-builder/admin/src/components/AllowedTypesSelect/MenuList.js (hunk 2 above)
Comment (soupette) [reply packages/strapi-plugin-content-type-builder/admin/src/components/AllowedTypesSelect/MenuList.js]: I have created new components

File: packages/strapi-plugin-content-type-builder/admin/src/components/AllowedTypesSelect/Text.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+
+const Text = styled.p`
```
Comment (soupette) [inline packages/strapi-plugin-content-type-builder/admin/src/components/AllowedTypesSelect/Text.js]: We will use the same `Text` as the one in the upload plugin but we should do this in another PR as we will use it in the entire content type builder.

File: packages/strapi-plugin-content-type-builder/admin/src/components/AllowedTypesSelect/MenuList.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+              <Label
+                htmlFor="overrideReactSelectBehavior"
+                onClick={() => {
+                  changeMediaAllowedTypes({
+                    target: { name: 'all', value: !areAllAllowedTypesSelected },
+                  });
+                }}
+              >
+                <SelectCheckbox
+                  id="checkAll"
```
Comment (virginieky) [inline packages/strapi-plugin-content-type-builder/admin/src/components/AllowedTypesSelect/MenuList.js]: Why do you need an id?
